internal/chainutil/chainutil.go

```diff
@@ -227,10 +245,54 @@ func (cs *ChainSim) MineBlock() types.Block {
 		for i := range txn.SiacoinInputs {
 			txn.SiacoinInputs[i].Signatures = []types.Signature{cs.privkey.SignHash(sigHash)}
 		}
-
 		txns = append(txns, txn)
 	}
-	cs.outputs = cs.outputs[:0]
+	cs.scOutputs = cs.scOutputs[:0]
+	for _, out := range cs.sfOutputs {
+		txn := types.Transaction{
+			SiafundInputs: []types.SiafundInput{{
+				Parent:      out,
+				SpendPolicy: types.PolicyPublicKey(cs.pubkey),
+			}},
+			SiafundOutputs: []types.SiafundOutput{
+				{Address: types.StandardAddress(cs.pubkey), Value: out.Value - 1},
+				{Address: types.Address{byte(cs.nonce >> 48), byte(cs.nonce >> 56), 1, 2, 3}, Value: 1},
+			},
+		}
+		sigHash := cs.Context.InputSigHash(txn)
+		for i := range txn.SiafundInputs {
+			txn.SiafundInputs[i].Signatures = []types.Signature{cs.privkey.SignHash(sigHash)}
+		}
+		txns = append(txns, txn)
+	}
+	cs.sfOutputs = cs.sfOutputs[:0]
+	for i, fc := range cs.contracts {
+		if (i % 2) == 0 {
+			// resolve
+			rev := fc.FileContract
+			rev.RevisionNumber = types.MaxRevisionNumber
+			txn := types.Transaction{
+				FileContractResolutions: []types.FileContractResolution{{
+					Parent:       fc,
+					Finalization: rev,
+				}},
+			}
+			txns = append(txns, txn)
+		} else {
+			// revise
+			rev := fc.FileContract
+			rev.Filesize += 5
```

Comment:
these revisions aren't being signed, so the transactions *should* be invalid. Why aren't the tests failing, then? 🤔

Reply:
I forget to initialize the contracts [here](https://github.com/SiaFoundation/siad/pull/101/commits/981da68f0df26c260f7954c37953659d8fab8d11#diff-609c8edd2177082e239bc3a17b95b57d53391c9ee8a73a46e2e531e31cd2b916R385).  Will be fixed shortly!